tools/windows_buildenv.bat

```diff
@@ -0,0 +1,104 @@
+@ECHO OFF
+SETLOCAL ENABLEDELAYEDEXPANSION
+
+CALL :REALPATH %~dp0\..
+SET MIXXX_ROOT=%RETVAL%
+
+IF "%PLATFORM%"=="" (
+    SET PLATFORM=x64
+)
+
+IF "%CONFIGURATION%"=="" (
+    SET CONFIGURATION=release-fastbuild
+)
+
+IF "%BUILDENV_BASEPATH%"=="" (
+    SET BUILDENV_BASEPATH="%MIXXX_ROOT%\buildenv"
+)
+
+IF "%BUILDENV_BASEPATH%"=="" (
+    SET BUILDENV_BASEPATH="%MIXXX_ROOT%\buildenv"
+)
+
+CALL :COMMAND_%1
+EXIT /B 0
+
+:COMMAND_name
+    CALL :READ_ENVNAME
+    ECHO "%RETVAL%"
+    IF "%CI%" == "true" (
+        ECHO BUILDENV_NAME=!RETVAL! >> %GITHUB_ENV%
+    )
+    GOTO :EOF
+
+:COMMAND_setup
+    CALL :READ_ENVNAME
+    SET BUILDENV_NAME=%RETVAL%
+    SET BUILDENV_PATH=%BUILDENV_BASEPATH%\%BUILDENV_NAME%
+    MD %BUILDENV_BASEPATH%
+
+    IF NOT EXIST %BUILDENV_PATH% (
+        SET BUILDENV_URL=https://downloads.mixxx.org/builds/buildserver/2.3.x-windows/%BUILDENV_NAME%.zip
+        ECHO Downloading !BUILDENV_URL!
+        BITSADMIN /transfer buildenvjob /download /priority normal !BUILDENV_URL! !BUILDENV_PATH!.zip
+        REM TODO: verify download using sha256sum?
+        ECHO Unpacking %BUILDENV_PATH%.zip
+        CALL :UNZIP "%BUILDENV_PATH%.zip" "%BUILDENV_BASEPATH%"
+        ECHO Unpacking complete.
+        DEL /f /q %BUILDENV_PATH%.zip
+    )
+
+    ECHO Using build environment: %BUILDENV_PATH%
+    ENDLOCAL
+
+    SET PATH=!BUILDENV_PATH!\bin;!PATH!
+    # Remove C:\Program Files\Git\usr\bin from the PATH
+    # This works around an issue on the windows-2016 builder provided by GitHub
+    # Actions, see this for details:
+    # - https://github.com/actions/cache/issues/333
+    # - https://github.com/actions/virtual-environments/issues/480
+    SET PATH=%PATH:C:\Program Files\Git\usr\bin;=%
+
+    FOR /D %%G IN (%BUILDENV_PATH%\Qt-*) DO (SET QT_PATH=%%G)
+    SET CMAKE_PREFIX_PATH=!BUILDENV_PATH!;!QT_PATH!
+
+    ECHO ^Environent Variables:
+    ECHO ^- PATH=!PATH!
+    ECHO ^CMake Configuration:
+    ECHO ^- CMAKE_PREFIX_PATH=!CMAKE_PREFIX_PATH!
+
+    IF NOT "%GITHUB_ENV%" == "" (
+        ECHO CMAKE_PREFIX_PATH=!CMAKE_PREFIX_PATH!>>%GITHUB_ENV%
+        ECHO PATH=!PATH!>>%GITHUB_ENV%
+    )
+    GOTO :EOF
+
+
+:UNZIP <newzipfile> <ExtractTo>
```

Comment:
:facepalm: that such a basic function is necessary.

Reply:
I didn't find another way to do this in batch without using external tools that might not be installed :shrug: